core/lib/via_btc_client/examples/data_inscription_example.rs

```diff
@@ -0,0 +1,682 @@
+
+    let sighash_type = EcdsaSighashType::All;
+
+    let fee_input_sighash = sighasher
+        .p2wpkh_signature_hash(
+            fee_input_index,
+            &fee_payer_utxo.1.script_pubkey,
+            fee_payer_utxo.1.value,
+            sighash_type,
+        )
+        .expect("failed to create sighash");
+
+    // Sign the sighash using the secp256k1 library (exported by rust-bitcoin).
+    let msg = Message::from(fee_input_sighash);
+    let fee_input_signature = secp.sign_ecdsa(&msg, &sk);
+
+    // Update the witness stack.
+    let fee_input_signature = bitcoin::ecdsa::Signature {
+        signature: fee_input_signature,
+        sighash_type,
+    };
+    let pk = sk.public_key(&secp);
+
+    *sighasher.witness_mut(fee_input_index).unwrap() = Witness::p2wpkh(&fee_input_signature, &pk);
+
+    // **Sign the reveal input**
+
+    let sighash_type = TapSighashType::All;
+    let prevouts = [fee_payer_utxo.1, reveal_input.1];
+    let prevouts = Prevouts::All(&prevouts);
+
+    let reveal_input_sighash = sighasher
+        .taproot_script_spend_signature_hash(
+            reveal_input_index,
+            &prevouts,
+            TapLeafHash::from_script(&inscription_script, LeafVersion::TapScript),
+            sighash_type,
+        )
+        .expect("failed to construct sighash");
+
+    // Sign the sighash using the secp256k1 library (exported by rust-bitcoin).
+    let msg = Message::from_digest(reveal_input_sighash.to_byte_array());
+    let reveal_input_signature = secp.sign_schnorr_no_aux_rand(&msg, &keypair);
+
+    // verify
+    secp.verify_schnorr(&reveal_input_signature, &msg, &internal_key)
+        .expect("signature is valid");
+
+    // Update the witness stack.
+    let reveal_input_signature = bitcoin::taproot::Signature {
+        signature: reveal_input_signature,
+        sighash_type,
+    };
+
+    let mut witness_data: Witness = Witness::new();
+
+    witness_data.push(&reveal_input_signature.to_vec());
+    witness_data.push(&inscription_script.to_bytes());
+
+    // add control block
+    witness_data.push(&reveal_input.2.serialize());
+
+    *sighasher
+        .witness_mut(reveal_input_index)
+        .ok_or("failed to get witness")
+        .unwrap() = witness_data;
+
+    let reveal_tx = sighasher.into_transaction();
+
+    // BOOM! Transaction signed and ready to broadcast.
+    println!("{:#?}", reveal_tx);
+
+    println!("reveal transaction: {:#?}", reveal_tx.raw_hex().to_string());
+
+    let txid = reveal_tx.compute_wtxid();
+    println!("reveal txid: {:?}", txid);
+
+    let txid = reveal_tx.compute_txid();
+    println!("reveal txid: {:?}", txid);
+}
+
+fn reveal_transaction_output_p2tr(
+    inscription_script: &ScriptBuf,
+    txid: &str,
+    taproot_spend_info: TaprootSpendInfo,
+) -> (OutPoint, TxOut, ControlBlock) {
+    let control_block = taproot_spend_info
+        .control_block(&(inscription_script.clone(), LeafVersion::TapScript))
+        .unwrap();
```

Comment:
usually "unwrap" and "expect" are not great for code. it can be used for quick implementation (and I think it works in case of small example)
but anyway, let's use error proper handling here
checkout `anyhow` crate. You could use anyhow::Result and anyhow::Context.
Error handling (even anyhow crate) always much better, than panics.

so here you can write
```
    let control_block = taproot_spend_info
        .control_block(&(inscription_script.clone(), LeafVersion::TapScript))?;
```